File: src/main/java/com/cjw/reactivecommunityproject/server/auth/service/AuthService.java
package com.cjw.reactivecommunityproject.server.auth.service;

import com.cjw.reactivecommunityproject.server.auth.model.AuthLoginVO;
import com.cjw.reactivecommunityproject.server.auth.model.AuthRegisterVO;

public interface AuthService {
    void register(AuthRegisterVO authRegisterVO, String salt);

    void login(AuthLoginVO authLoginVO);
}

File: src/main/kotlin/com/devcjw/reactivecommunity/auth/service/AuthService.kt
package com.devcjw.reactivecommunity.auth.service

import com.devcjw.reactivecommunity.auth.model.domain.*
import com.devcjw.reactivecommunity.common.model.RestResponseVO
import reactor.core.publisher.Mono

interface AuthService {
    fun login(loginDTO: ReqAuthLoginVO): Mono<RestResponseVO<RepAuthTokenVO>>
    fun signup(reqAuthSignupVO: ReqAuthSignupVO): Mono<RestResponseVO<Void>>

    fun check(reqAuthCheckVO: ReqAuthCheckVO): Mono<RestResponseVO<Void>>
    fun reissue(reqAuthReissueVO: ReqAuthReissueVO): Mono<RestResponseVO<RepAuthReissueVO>>

}